.github/workflows/build-and-deploy-notifications.yml

```diff
@@ -0,0 +1,65 @@
+name: Build and Deploy
+
+on:
+  push:
+    branches:
+    - main
+    - deploy-resources
+  pull_request:
+    branches:
+    - main
+
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: Build & test
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: '6.0.x'
+
+    - name: Lint
+      run: |
+        #https://github.com/dotnet/format/issues/1433
+        dotnet tool install -g dotnet-format --version "6.*" --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json
+        dotnet-format --verify-no-changes
+      working-directory: DqtNotifications
+
+    - name: Build
+      run: dotnet build --configuration Release
+      working-directory: DqtNotifications
+
+  deploy:
+    name: Deploy to ${{ matrix.environment }} environment
+    runs-on: ubuntu-latest
+    #if: github.ref == 'refs/heads/main' && github.event_name == 'push'
```

Comment:
Could we set this up like Identity with different jobs/triggers for each environment? https://github.com/DFE-Digital/get-an-identity/blob/main/.github/workflows/build.yml#L194

Reply:
We can but that approach creates a lot of unnecessary duplication, matrix deployments are the normal approach in other Teachers projects.  The way the get-an-identity workflow is set up also creates a risk of accidently deploying a branch to production, I'm looking at some improvements we could make to that this morning

Reply:
resolved